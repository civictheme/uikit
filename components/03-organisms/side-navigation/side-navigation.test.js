const template = 'components/03-organisms/side-navigation/side-navigation.twig';

describe('Side Navigation Component', () => {
  test('renders with only required attributes', async () => {
    const c = await dom(template, {
      items: [
        { title: 'Home', url: '/' },
      ],
    });

    expect(c.querySelectorAll('.ct-side-navigation')).toHaveLength(1);

    const menuItems = c.querySelectorAll('.ct-side-navigation__menu .ct-link');
    expect(menuItems).toHaveLength(1);
    expect(menuItems[0].textContent.trim()).toBe('Home');

    assertUniqueCssClasses(c);
  });

  test('renders with all attributes provided', async () => {
    const c = await dom(template, {
      theme: 'dark',
      items: [
        { title: 'Home', url: '/' },
        { title: 'About', url: '/about' },
        { title: 'Services', url: '/services' },
        { title: 'Contact', url: '/contact' },
      ],
      title: 'Main Navigation',
      vertical_spacing: 'both',
      attributes: 'data-test="true"',
      modifier_class: 'additional-class',
    });

    expect(c.querySelectorAll('.ct-side-navigation')).toHaveLength(1);
    expect(c.querySelector('.ct-side-navigation').classList).toContain('ct-theme-dark');
    expect(c.querySelector('.ct-side-navigation').classList).toContain('ct-vertical-spacing-inset--both');
    expect(c.querySelector('.ct-side-navigation').classList).toContain('additional-class');

    const menuItems = c.querySelectorAll('.ct-side-navigation__menu .ct-link');
    expect(menuItems).toHaveLength(4);
    expect(menuItems[0].textContent.trim()).toBe('Home');
    expect(menuItems[1].textContent.trim()).toBe('About');
    expect(menuItems[2].textContent.trim()).toBe('Services');
    expect(menuItems[3].textContent.trim()).toBe('Contact');

    expect(c.querySelector('.ct-side-navigation__title').textContent.trim()).toBe('Main Navigation');
    expect(c.querySelector('.ct-side-navigation').getAttribute('data-test')).toBe('true');

    assertUniqueCssClasses(c);
  });

  test('renders with some attributes missing', async () => {
    const c = await dom(template, {
      theme: 'light',
      items: [
        { title: 'Home', url: '/' },
        { title: 'About', url: '/about' },
      ],
      title: '',
      vertical_spacing: '',
      modifier_class: '',
    });

    expect(c.querySelectorAll('.ct-side-navigation')).toHaveLength(1);
    expect(c.querySelector('.ct-side-navigation').classList).toContain('ct-theme-light');

    const menuItems = c.querySelectorAll('.ct-side-navigation__menu .ct-link');
    expect(menuItems).toHaveLength(2);
    expect(menuItems[0].textContent.trim()).toBe('Home');
    expect(menuItems[1].textContent.trim()).toBe('About');

    expect(c.querySelector('.ct-side-navigation__title')).toBeNull();

    assertUniqueCssClasses(c);
  });

  test('does not render when items are empty', async () => {
    const c = await dom(template, {
      items: [],
    });

    expect(c.querySelectorAll('.ct-side-navigation')).toHaveLength(0);
  });
});
