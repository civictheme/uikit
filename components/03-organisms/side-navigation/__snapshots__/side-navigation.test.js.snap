// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Navigation Component does not render when items are empty 1`] = `
<div>
  


</div>
`;

exports[`Side Navigation Component renders with all attributes provided 1`] = `
<div>
  

  
  <div
    class="ct-side-navigation ct-theme-dark ct-vertical-spacing-inset--both additional-class"
    data-test="true"
  >
    
    


  
    <div
      class="ct-skip-link ct-theme-dark "
    >
      
    


  
  
  
  
      <a
        class="ct-link ct-theme-dark     ct-skip-link__link ct-visually-hidden ct-focusable"
        href="#main-content"
        title="Skip to main content"
      >
            Skip to main content  
      </a>
      
  
    </div>
    
                  


  
    <h2
      class="ct-heading ct-theme-dark ct-side-navigation__title"
    >
      Main Navigation
    </h2>
    
              






            
    <ul
      class="ct-menu ct-menu--level-0 ct-theme-dark ct-side-navigation__menu"
      data-component-name="ct-menu"
    >
      
    
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-dark     ct-menu__item__link"
          href="/"
          title="Home"
        >
              Home  
        </a>
        

        
        
      
      </li>
      
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-dark     ct-menu__item__link"
          href="/about"
          title="About"
        >
              About  
        </a>
        

        
        
      
      </li>
      
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-dark     ct-menu__item__link"
          href="/services"
          title="Services"
        >
              Services  
        </a>
        

        
        
      
      </li>
      
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-dark     ct-menu__item__link"
          href="/contact"
          title="Contact"
        >
              Contact  
        </a>
        

        
        
      
      </li>
      
    
          
    </ul>
    
      
  
  </div>
  

</div>
`;

exports[`Side Navigation Component renders with only required attributes 1`] = `
<div>
  

  
  <div
    class="ct-side-navigation ct-theme-light  "
  >
    
    


  
    <div
      class="ct-skip-link ct-theme-light "
    >
      
    


  
  
  
  
      <a
        class="ct-link ct-theme-light     ct-skip-link__link ct-visually-hidden ct-focusable"
        href="#main-content"
        title="Skip to main content"
      >
            Skip to main content  
      </a>
      
  
    </div>
    
        






            
    <ul
      class="ct-menu ct-menu--level-0 ct-theme-light ct-side-navigation__menu"
      data-component-name="ct-menu"
    >
      
    
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-light     ct-menu__item__link"
          href="/"
          title="Home"
        >
              Home  
        </a>
        

        
        
      
      </li>
      
    
          
    </ul>
    
      
  
  </div>
  

</div>
`;

exports[`Side Navigation Component renders with some attributes missing 1`] = `
<div>
  

  
  <div
    class="ct-side-navigation ct-theme-light  "
  >
    
    


  
    <div
      class="ct-skip-link ct-theme-light "
    >
      
    


  
  
  
  
      <a
        class="ct-link ct-theme-light     ct-skip-link__link ct-visually-hidden ct-focusable"
        href="#main-content"
        title="Skip to main content"
      >
            Skip to main content  
      </a>
      
  
    </div>
    
        






            
    <ul
      class="ct-menu ct-menu--level-0 ct-theme-light ct-side-navigation__menu"
      data-component-name="ct-menu"
    >
      
    
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-light     ct-menu__item__link"
          href="/"
          title="Home"
        >
              Home  
        </a>
        

        
        
      
      </li>
      
          
                  
      
      <li
        class="ct-menu__item ct-menu__item--level-0   "
      >
        

        


  
  
  
  
        <a
          class="ct-link ct-theme-light     ct-menu__item__link"
          href="/about"
          title="About"
        >
              About  
        </a>
        

        
        
      
      </li>
      
    
          
    </ul>
    
      
  
  </div>
  

</div>
`;
